### PR TITLE
feat: improve output for global update

### LINKED ISSUE #2345: Show the version updates in `pixi global update`
### Problem description

Is it possible to show the version updates after running the `pixi global update` command, just like it was in the old `pixi global upgrade-all` command? It was a very convenient way to keep track of the changes.

### PR DESCRIPTION
**Pull request by nichmor**
expected results are described here:
https://github.com/prefix-dev/pixi/pull/2284

Output when env is up-to-date:
image

Output when only transient dep is updated:
image

Output when only one package is updated:
image

Output when multiple top-level dependencies have been updated:
Screenshot 2024-10-22 at 17 09 07

combined output when updating multiple envs:

image

with version update:

image

- [x] add tests

Fixes #2345

### CODE REVIEW

**Comment by ruben-arts** [comment]
Could you also add the version of the packages that we're changed?

**Comment by nichmor** [comment]
> Could you also add the version of the packages that we're changed?

sure - how do you want to see it?

`Updated package ripgrep 10 -> 12` ?

and the same for multiple packages?

File: src/global/common.rs [hunk 1]
```diff
@@ -257,6 +259,104 @@ pub(crate) struct EnvChanges {
     pub changes: HashMap<EnvironmentName, EnvState>,
 }
 
+#[derive(Debug, Default)]
+pub(crate) struct InstallChanges {
+    pub changes: HashMap<PackageName, InstallChange>,
+}
```
Comment (Hofer-Julian) [inline src/global/common.rs:265]: I would expect this functionality to be part of `StateChanges`

**Review by Hofer-Julian — COMMENTED** [review]
The versions should be blue, similar to `pixi global list`

image

File: src/global/common.rs (hunk 1 above)
Comment (nichmor) [reply src/global/common.rs]: as discussed, moved it to StateChanges

**Review by Hofer-Julian — COMMENTED** [review]
I had a few minor comments, but apart from that it looks good to me!

File: src/global/common.rs [hunk 2]
```diff
@@ -560,6 +685,67 @@ pub fn check_all_exposed(
     auto_exposed
 }
 
+pub(crate) fn get_install_changes(
+    install_transaction: Transaction<PrefixRecord, RepoDataRecord>,
+) -> HashMap<PackageName, InstallChange> {
+    let mut install_changes: HashMap<PackageName, InstallChange> = HashMap::default();
+
+    for transaction in install_transaction.operations {
```
[the remark this answers is not in the document]
Comment (nichmor) [reply src/global/common.rs:693]: changed it to into_iter.for_each

File: tests/integration/conftest.py [hunk 3]
```diff
@@ -31,3 +31,8 @@ def global_update_channel_1(test_data: Path) -> str:
 @pytest.fixture
 def non_self_expose_channel(test_data: Path) -> str:
     return test_data.joinpath("channels", "non_self_expose_channel").as_uri()
+
+
+@pytest.fixture
+def non_self_expose_channel_2(test_data: Path) -> str:
```
[the remark this answers is not in the document]
Comment (nichmor) [reply tests/integration/conftest.py:37]: renamed also the recipe

**Review by Hofer-Julian — APPROVED** [review]

**Pull request merged**